### PR TITLE
implemented backward for Pad2D & test

### PR DESCRIPTION
**Pull request by ziofil**

### CODE REVIEW

File: tinygrad/ops.py [hunk 1]
```diff
@@ -97,13 +97,15 @@ def backward(ctx, grad_output):
...
+    ctx.save_for_backward(padding)
     return np.pad(x,
       ((0,0), (0,0),
        (padding[0], padding[1]), (padding[2], padding[3])))
 
   @staticmethod
   def backward(ctx, grad_output):
-    raise Exception("write this")
+    padding, = ctx.saved_tensors
+    return grad_output[..., padding[0]:-padding[1], padding[2]:-padding[3]]
```
Comment (geohot) [inline tinygrad/ops.py:108]: You can do ctx.padding even if you don't save it.

File: test/test_ops.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def test_logsoftmax(self):
     helper_test_op([(45,65)], lambda x: torch.nn.LogSoftmax(dim=1)(x), Tensor.logsoftmax, atol=1e-5, gpu=self.gpu)
 
   def test_pad2d(self):
-    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (1,1,1,1)), lambda x: x.pad2d(padding=(1,1,1,1)), gpu=self.gpu, forward_only=True)
+    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (1,1,1,1)), lambda x: x.pad2d(padding=(1,1,1,1)), gpu=self.gpu)
```
Comment (geohot) [inline test/test_ops.py:67]: forward_only=self.gpu ?
